pkg/metricgen/intake.go

```diff
@@ -0,0 +1,86 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package metricgen
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+
+	"go.elastic.co/apm/module/apmotel/v2"
+	"go.elastic.co/apm/v2"
+	"go.opentelemetry.io/otel/sdk/metric"
+)
+
+// SendIntakeV2 sends specific metrics to the configured Elastic APM intake V2.
+//
+// Metrics sent are:
+// - apm(float64, value=1.0); gathered from a apm.MetricGatherer
+// - apmotel(float64, value=1.0); gathered from a otel MeterProvider through apmotel bridge
+// All builtin APM Agent metrics have been disabled.
+func SendIntakeV2(ctx context.Context, cfg Config) (EventStats, error) {
```

Comment:
This means the syntaxhas to be `SendIntakeV2(ctx, NewConfig(WithElasticAPMServiceName("name"))`, which seems quite verbose.
Does the config need to be exposed? How about passing only options and making the config struct private?
```
SendIntakeV2(ctx, WithElasticAPMServiceName("name"))
```